common/src/main/java/com/wynntils/commands/LootrunCommand.java

```diff
@@ -0,0 +1,430 @@
+                e.printStackTrace();
+            }
+        }
+        return 1;
+    }
+
+    private int recordLootrun(CommandContext<CommandSourceStack> context) {
+        LootrunUtils.register();
+        if (LootrunUtils.recording == null) {
+            LootrunUtils.recording = new LootrunUncompiled(new ArrayList<>(), new HashSet<>(), new ArrayList<>(), null);
+            LootrunUtils.recordingInformation = new RecordingInformation();
+            context.getSource()
+                    .sendSuccess(
+                            new TranslatableComponent(
+                                    "feature.wynntils.lootrunUtils.recordStart",
+                                    new TextComponent("/lootrun record")
+                                            .withStyle(ChatFormatting.UNDERLINE)
+                                            .withStyle((style) -> style.withClickEvent(
+                                                    new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/lootrun record")))),
+                            false);
+        } else {
+            LootrunUtils.lootrun = LootrunUtils.recordingCompiled;
+            LootrunUtils.uncompiled = LootrunUtils.recording;
+            LootrunUtils.recording = null;
+            LootrunUtils.recordingCompiled = null;
+            LootrunUtils.recordingInformation = null;
+            context.getSource()
+                    .sendSuccess(
+                            new TranslatableComponent(
+                                            "feature.wynntils.lootrunUtils.recordStop1",
+                                            new TextComponent("/lootrun clear")
+                                                    .withStyle(ChatFormatting.UNDERLINE)
+                                                    .withStyle((style) -> style.withClickEvent(new ClickEvent(
+                                                            ClickEvent.Action.RUN_COMMAND, "/lootrun clear"))))
+                                    .withStyle(ChatFormatting.RED)
+                                    .append("\n")
+                                    .append(new TranslatableComponent(
+                                                    "feature.wynntils.lootrunUtils.recordStop2",
+                                                    new TextComponent("/lootrun save <name>")
+                                                            .withStyle(ChatFormatting.UNDERLINE)
+                                                            .withStyle((style) -> style.withClickEvent(new ClickEvent(
+                                                                    ClickEvent.Action.SUGGEST_COMMAND,
+                                                                    "/lootrun save "))))
+                                            .withStyle(ChatFormatting.GREEN)),
+                            false);
+        }
+        return 1;
+    }
+
+    private int saveLootrun(CommandContext<CommandSourceStack> context) {
```

Comment:
Seems like most of this file should be in a Feature, rather than in a Command, yes?

Reply:
honestly i think a lot of the stuff in the command class should go in `LootrunUtils`, and a lot of stuff currently in `LootrunUtils` could go into a `LootrunFeature` class, to align with our current convention of doing event handling in Features

Reply:
a `LootrunFeature` wouldn't be too substantial, but i do think there's value in keeping our event handling standardized - even if it's just calling util methods from the event methods. additionally, if we make our configuration framework group by Feature classes (annotated variables, maybe) it would be best to have a place for lootrun options to go

Reply:
For me the Commands should be just entries into functionality, a conversion layer between the chat prompt and the functionality.

See it like this. If we were to add a button to the ingame menu "Start lootrun", should that call into a Command class? That seems wrong. (In fact, I think we have some kind of load lootrun selection GUI in legacy..?)

Once again, I feel we're missing a place to put "Wynntils functionality", that is not a direct part of a feature. Could be have a `functionality` package, as a peer to `features`, for Wynntils functionality that is not bundled as a feature to be turned on and off? Otherwise I think the best solution still is to shoehorn the lootruns into a Feature.

Reply:
And interesting idea with Features turned off and on by command. What if we had a `LootrunManagerFeature` that handles lootrun management (loading, saving, deleting and --which is missing in legacy -- editing), and a `LootrunRecording` and `LootrunPlayback`/`LootrunDisplaying` that are enabled, temporary, when saving or playing back/showing a lootrun?

Would that be a good idea, or just confuse things?

Reply:
I feel like the `LootrunManagerFeature` is not really a feature, cause it does not do anything by itself. It can not be "enabled" or "disabled", I don't feel like there is anything to disable.

As for `LootrunRecordingFeature` and `LootrunPlaybackFeature`, they do feel more connected to features, but then they are not really like any other feature, where it is clear what enabling and disabling them does. 

I think it may be the best option to invent something like the aforementioned "functionality", although I don't have solid ideas on how it would look like.